deepcell/image_generators/__init__.py

```diff
@@ -173,9 +174,9 @@ def _transform_masks(y, transform, data_format=None, **kwargs):
         beta = kwargs.pop('beta', 1)
 
         if data_format == 'channels_first':
-            y_transform = np.zeros(tuple([y.shape[0]] + list(y.shape[2:])))
+            y_transform = np.zeros(tuple([y.shape[0]] + list(y.shape[2:])), dtype='float16')
         else:
-            y_transform = np.zeros(y.shape[0:-1])
+            y_transform = np.zeros(y.shape[0:-1], dtype='float16')
```

Comment:
why are these float16, but later we re-cast to uint8?

Should we have an optional kwarg for the target dtype? It can default to uint8 but will allow a bit more customization.

Reply:
I wanted to pick the smallest dtype appropriate for that location. I used float16 since the inner distance transform gives floats, but then if it get binned and changed to one-hot, it can be converted to ints